sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TryStatementsWithIdenticalCatchShouldBeMerged.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        void Simple()
+        {
+            try { }
+            catch (Exception)
+            {
+            }
+            finally { }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 9.}}
+            catch (Exception)
+            {
+            }
+            finally { }
+
+            try { }
+            finally { }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 15.}}
```

Comment:
The rule says it applies "even if there is intervening code outside the _try_ block". In this case (and in the case in line 30) the intervening code include a _try-catch_ block.
The rule isn't explicit about whether the intervening code must be "normal" code or whether it can include _try-catch_ block. What was the intention of the RSPEC?

Currently, this method is recommending merging blocks (2 and 4) and (3 and 5), which probably isn't possible. Do you think that matters? I think it probably doesn't - once the user has merged one block, the other blocks probably won't be mergeable.

Reply:
- Regarding the first question - I think it doesn't matter what's between the two try blocks.
```
try {} // #1

// whatever is here, even other try blocks

try {} // #2 should be mergeable with #1 even if there are other try blocks in between
```

- It shouldn't matter imo. If you merge two of the blocks the third one will not be mergeable unless it still has the same catch clauses.